vm/rust/src/jsonrpc.rs

```diff
@@ -185,12 +185,53 @@ use blockifier::execution::call_info::OrderedEvent as BlockifierOrderedEvent;
 impl From<BlockifierOrderedEvent> for OrderedEvent {
     fn from(val: BlockifierOrderedEvent) -> Self {
         OrderedEvent {
-            order: Some(val.order),
+            order: val.order,
             event: val.event,
         }
     }
 }
 
+#[derive(Serialize)]
+pub struct ExecutionResources {
+    pub steps: usize,
+    pub memory_holes: usize,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub range_check_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pedersen_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub poseidon_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ec_op_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ecdsa_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub bitwise_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub keccak_builtin_applications: Option<usize>,
+    // https://github.com/starkware-libs/starknet-specs/pull/167
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub segment_arena_builtin: Option<usize>,
+}
```

Comment:
Also, these must be marshaled in to a hex string if json is legacy :/

Reply:
That's a nice catch

Reply:
I think I got it confused sorry:/ Trace API v0.5 doesnt have execution resources, so there is no need to have the ability to marshal these into hex strings.

But we do need to support that for receipts.